ampd/src/stacks/http_client.rs

```diff
@@ -134,9 +159,23 @@ impl Client {
             .get(endpoint.clone())
             .send()
             .await
+            .inspect_err(|_| {
```

Comment:
`transaction` is a private function that is only called by `valid_transaction` (and your test, which shouldn't call into a private function). If you record the metrics in `valid_transaction` you only have to define it once